File: php/Motorcycle.php
<?php

require_once "Vehicle.php";

class Motorcycle extends Vehicle {
    private $type;
    private $tankCapacity;

    // Constructor
    public function __construct($plate, $brand, $year, $clr, $type, $tankCapacity) {
        parent::__construct($plate, $brand, $clr, $year);
        $this->type = $type;
        $this->tankCapacity = $tankCapacity;
    }

    // Getters
    public function getType() {
        return $this->type;
    }

    public function getTankCapacity() {
        return $this->tankCapacity;
    }

    // Setters
    public function setType($type) {
        $this->type = $type;
    }

    public function setTankCapacity($capacity) {
        $this->tankCapacity = $capacity;
    }
}

File: java/Motorcycle.java
public class Motorcycle extends Vehicle {
    private int type;
    private int tankCapacity;

    // Constructor
    public Motorcycle(String plate, String brand, int year, String clr, int type, int tankCapacity) {
        super(plate, brand, clr, year);
        this.type = type;
        this.tankCapacity = tankCapacity;
    }

    // Getters
    public int getType() {
        return type;
    }

    public int getTankCapacity() {
        return tankCapacity;
    }

    // Setters
    public void setType(int type) {
        this.type = type;
    }

    public void setTankCapacity(int capacity) {
        this.tankCapacity = capacity;
    }
}
